packages/camel-catalog/catalog-generator/src/main/java/io/kaoto/camelcatalog/KaotoCamelCatalogBuilder.java

```diff
@@ -0,0 +1,403 @@
+                            var jsonGenerator = new JsonFactory().createGenerator(writer).useDefaultPrettyPrinter();
+                            jsonMapper.writeTree(jsonGenerator, catalogNode);
+                            catalog = writer.toString();
+                        }
+
+                        var outputFileName = String.format(
+                                "%s-%s-%s.json", CAMEL_CATALOG_AGGREGATE, name, Util.generateHash(catalog));
+                        var output = outputDirectory.toPath().resolve(outputFileName);
+                        Files.writeString(output, catalog);
+                        var indexEntry = new Entry(
+                                name,
+                                "Aggregated Camel catalog for " + name,
+                                cameCatalogVersion,
+                                outputFileName);
+                        index.getCatalogs().put(name, indexEntry);
+                    } catch (Exception e) {
+                        LOGGER.log(Level.SEVERE, e.toString(), e);
+                    }
+                });
+            } catch (Exception e) {
+                LOGGER.log(Level.SEVERE, e.toString(), e);
+            }
+        }
+
+        private void processKameletBoundaries(Index index) {
+            if (camelCatalogVersionLoader.getKameletBoundaries().isEmpty()) {
+                LOGGER.severe(String.format(
+                        "Kamelet boundaries are not loaded"));
+                return;
+            }
+
+            var indexEntry = getKameletsEntry(camelCatalogVersionLoader.getKameletBoundaries(), KAMELET_BOUNDARIES_KEY,
+                    KAMELET_BOUNDARIES_FILENAME, "Aggregated Kamelet boundaries in JSON");
+            index.getCatalogs().put(indexEntry.name(), indexEntry);
+        }
+
+        private void processKamelets(Index index) {
+            if (camelCatalogVersionLoader.getKamelets().isEmpty()) {
+                LOGGER.severe(String.format(
+                        "Kamelets are not loaded"));
+                return;
+            }
+
+            var indexEntry = getKameletsEntry(camelCatalogVersionLoader.getKamelets(), KAMELETS, KAMELETS_AGGREGATE,
+                    "Aggregated Kamelets in JSON");
+            index.getCatalogs().put(indexEntry.name(), indexEntry);
+        }
+
+        private Entry getKameletsEntry(List<String> kamelets, String name, String filename, String description) {
+            var root = jsonMapper.createObjectNode();
+
+            try {
+                kamelets.forEach(kamelet -> {
+                    processKameletFile(kamelet, root);
+                });
+
+                JsonFactory jsonFactory = new JsonFactory();
+                var outputStream = new ByteArrayOutputStream();
+                var writer = new OutputStreamWriter(outputStream);
+                var jsonGenerator = jsonFactory.createGenerator(writer).useDefaultPrettyPrinter();
```

Comment:
the createGenrator is supposed to be closed, so either use a try with resource or close it